tests/pytests/unit/grains/test_core.py

```diff
@@ -1174,160 +1182,154 @@ def test_bsd_memdata():
-                    with patch.object(os.path, "exists", path_exists_mock):
-                        with patch("salt.utils.path.which") as mock:
-                            mock.return_value = "/sbin/sysctl"
-                            # Make a bunch of functions return empty dicts,
-                            # we don't care about these grains for the
-                            # purposes of this test.
-                            with patch.object(core, "_bsd_cpudata", empty_mock):
-                                with patch.object(core, "_hw_data", empty_mock):
-                                    with patch.object(core, "_virtual", empty_mock):
-                                        with patch.object(core, "_ps", empty_mock):
-                                            # Mock the osarch
-                                            with patch.dict(
-                                                core.__salt__,
-                                                {"cmd.run": cmd_run_mock},
-                                            ):
-                                                os_grains = core.os_data()
+            os_grains = core.os_data()
 
     assert os_grains.get("mem_total") == 2023
     assert os_grains.get("swap_total") == 400
 
 
-@pytest.mark.skip_on_windows(reason="System is Windows")
-def test_docker_virtual():
+@pytest.mark.skip_on_windows
+@pytest.mark.parametrize(
+    "cgroup_substr",
+    (
+        ":/system.slice/docker",
+        ":/docker/",
+        ":/docker-ce/",
+    ),
+)
+def test_docker_virtual(cgroup_substr):
     """
     Test if virtual grains are parsed correctly in Docker.
     """
-    with patch.object(os.path, "isdir", MagicMock(return_value=False)):
-        with patch.object(
-            os.path,
-            "isfile",
-            MagicMock(side_effect=lambda x: True if x == "/proc/1/cgroup" else False),
-        ):
-            for cgroup_substr in (
-                ":/system.slice/docker",
-                ":/docker/",
-                ":/docker-ce/",
-            ):
-                cgroup_data = "10:memory{}a_long_sha256sum".format(cgroup_substr)
-                log.debug("Testing Docker cgroup substring '%s'", cgroup_substr)
-                with patch("salt.utils.files.fopen", mock_open(read_data=cgroup_data)):
-                    with patch.dict(core.__salt__, {"cmd.run_all": MagicMock()}):
-                        grains = core._virtual({"kernel": "Linux"})
-                        assert grains.get("virtual_subtype") == "Docker"
-                        assert grains.get("virtual") == "container"
+    cgroup_data = "10:memory{}a_long_sha256sum".format(cgroup_substr)
+    log.debug("Testing Docker cgroup substring '%s'", cgroup_substr)
+    with patch.object(os.path, "isdir", MagicMock(return_value=False)), patch.object(
+        os.path,
+        "isfile",
+        MagicMock(side_effect=lambda x: True if x == "/proc/1/cgroup" else False),
```

Comment:
**suggestion** another set of extra MagicMock declarations (the run_all one is valid though)